### PR TITLE
bpo-46672: fix `NameError` in `asyncio.gather` if type check fails

### PR DESCRIPTION
**Pull request by sobolevn**
I am a bit unsure about the full coverage of `if outer is None or outer.done():` condition. Does it cover all corner cases?

https://bugs.python.org/issue46672

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2022-02-07-13-15-16.bpo-46672.4swIjx.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixes ``NameError`` in :func:`asyncio.gather` when initial type check fails.
```
Comment (AlexWaygood) [inline Misc/NEWS.d/next/Library/2022-02-07-13-15-16.bpo-46672.4swIjx.rst]: Nit
```suggestion
Fix ``NameError`` in :func:`asyncio.gather` when initial type check fails.
```

File: Lib/asyncio/tasks.py [hunk 2]
```diff
@@ -634,7 +634,7 @@ def _ensure_future(coro_or_future, *, loop=None):
         loop = events._get_event_loop(stacklevel=4)
     try:
         return loop.create_task(coro_or_future)
-    except RuntimeError: 
+    except RuntimeError:
```
Comment (sobolevn) [inline Lib/asyncio/tasks.py:637]:
```suggestion
    except RuntimeError: 
```

Comment (sobolevn) [reply Lib/asyncio/tasks.py:637]: Probably this was modified by my IDE without asking me.
Click on `Commit suggestion` to revert. Or let it be!

**Review by kumaraditya303 — APPROVED** [review]
LGTM

File: Lib/asyncio/tasks.py (hunk 2 above)
Comment (JelleZijlstra) [reply Lib/asyncio/tasks.py:637]: Let's just fix this.

**Review by JelleZijlstra — APPROVED** [review]
Confirmed that the new test fails without the fix.

**Review by asvetlov — APPROVED** [review]

**Pull request merged**